server/coordinator.go

```diff
@@ -165,27 +166,63 @@ func (c *coordinator) run() {
 	}
 	log.Info("coordinator starts to run schedulers")
 
-	k := 0
+	scheduleNames, configs, err := c.cluster.storage.LoadAllScheduleConfig()
+	if err != nil {
+		log.Fatal("cannot load schedulers' config")
+	}
+
 	scheduleCfg := c.cluster.opt.Load().Clone()
+	// The new way to create schedule with the independent configuration.
+	for i, name := range scheduleNames {
+		data := configs[i]
+		typ := schedule.FindSchedulerTypeByName(name)
+		var cfg config.SchedulerConfig
+		for _, c := range scheduleCfg.Schedulers {
+			if c.Type == typ {
+				cfg = c
+				break
+			}
+		}
+		if len(cfg.Type) == 0 {
+			log.Error("can not create scheduler with independent configuration", zap.String("scheduler-name", name))
+			continue
+		}
+		if cfg.Disable {
+			log.Info("skip create scheduler with independent configuration", zap.String("scheduler-name", name), zap.String("scheduler-type", cfg.Type))
+			continue
+		}
+		s, err := schedule.CreateScheduler(cfg.Type, c.opController, c.cluster.storage, schedule.ConfigJSONDecoder([]byte(data)))
+		if err != nil {
+			log.Error("can not create scheduler with independent configuration", zap.String("scheduler-name", name), zap.Error(err))
```

Comment:
It's hard to tell the difference between this line and line 186 if an error happens.